mlflow/store/dbmodels/models.py

```diff
@@ -188,6 +189,47 @@ def to_mlflow_entity(self):
         return Run(run_info=run_info, run_data=run_data)
 
 
+class SqlExperimentTag(Base):
+    """
+    DB model for :py:class:`mlflow.entities.RunTag`.
+    These are recorded in ``experiment_tags`` table.
+    """
+    __tablename__ = 'experiment_tags'
+
+    key = Column(String(250))
+    """
+    Tag key: `String` (limit 250 characters). *Primary Key* for ``tags`` table.
+    """
+    value = Column(String(250), nullable=True)
```

Comment:
As a test, I attempted to log an experiment tag with a 500-byte value using the current configuration against MySQL. Here's the code I ran:
```
import mlflow
from mlflow.tracking.client import MlflowClient

mlflow.set_tracking_uri("mysql://root:password@localhost:33060/exptags")

client = MlflowClient()

mlflow.set_experiment("Dogface")
exp = client.get_experiment("1")
print(exp)

long_value = "a" * 500
print(len(long_value))

client.set_experiment_tag(
    "1",
    "key",
    long_value)

longtag = client.get_experiment("1").tags["key"]
print(len(longtag))
```

Interestingly, when running this code against a fresh database seemed to truncate the tag; here was the STDOUT:
```
$ python sqllong.py
2019/07/19 14:24:48 INFO mlflow.store.sqlalchemy_store: Creating initial MLflow database tables...
2019/07/19 14:24:48 INFO mlflow.store.db.utils: Updating database tables at mysql://root:password@localhost:33060/exptags
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 451aebb31d03, add metric step
INFO  [alembic.runtime.migration] Running upgrade 451aebb31d03 -> 90e64c465722, migrate user column to tags
INFO  [alembic.runtime.migration] Running upgrade 90e64c465722 -> 181f10493468, allow nulls for metric values
INFO  [alembic.runtime.migration] Running upgrade 181f10493468 -> df50e92ffc5e, Add Experiment Tags Table
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO: 'Dogface' does not exist. Creating a new experiment
<Experiment: artifact_location='./mlruns/1', experiment_id='1', lifecycle_stage='active', name='Dogface', tags={}>
500
250
```

However, attempting to run the exact same code again yields the following error output:
```
 $ python sqllong.py
<Experiment: artifact_location='./mlruns/1', experiment_id='1', lifecycle_stage='active', name='Dogface', tags={'key': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'}>
500
Traceback (most recent call last):
  File "sqllong.py", line 21, in <module>
    long_value)
  File "/Users/czumar/mlflow/mlflow/tracking/client.py", line 187, in set_experiment_tag
    self.store.set_experiment_tag(exp_id, tag)
  File "/Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py", line 492, in set_experiment_tag
    value=tag.value))
  File "/Users/czumar/anaconda2/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py", line 156, in make_managed_session
    raise MlflowException(message=e, error_code=INTERNAL_ERROR)
mlflow.exceptions.MlflowException: (MySQLdb._exceptions.DataError) (1406, "Data too long for column 'value' at row 1")
[SQL: UPDATE experiment_tags SET value=%s, experiment_id=%s WHERE experiment_tags.`key` = %s AND experiment_tags.experiment_id = %s]
[parameters: ('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ... (202 characters truncated) ... aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', '1', 'key', 1)]
(Background on this error at: http://sqlalche.me/e/9h9h)
```

We should figure out what's causing this behavioral inconsistency and fix it. Can you also try to reproduce the problem with run tags?

Reply:
Per offline discussion, this seems related to strict mode execution on the SQL queries. Solution is to do tag validation before setting.